lib/filters.js

```diff
@@ -14,57 +14,74 @@ exports.tasks_by_id = function(doc, req) {
 };
 
 exports.doc_by_place = function(doc, req) {
-  var ok = function(place) {
-    if (!req.query.id) {
-      // admin
-      return true;
+  var ALL = "all",
+      UNASSIGNED = "unassigned",
+      ADMIN = "admin";
+
+  function usersThatCanSeeThisDoc(doc) {
```

Comment:
The idea is that this core function outputs a collection of keys useful in both situations. In the filter you `indexOf` over them, in the view you'd emit them.